### ISSUE TITLE
fast-glob doesn't match list segments correctly

### ISSUE THREAD

**Issue by AgentEnder** [mrmlnc/fast-glob#365]
### Environment

  * OS Version: MacOS Big Sur (11.6.4)
  * Node.js Version: 17

### Actual behavior
List-segments are matched in a way that is compatible with `glob`

### Expected behavior
Some matches are not reported properly when using glob patterns containing quotes

### Steps to reproduce
- See linked PR with test: #366

**Comment by AgentEnder** [comment]
@mrmlnc When checking this out in a bit more detail, it seems like when matching list segments the `globstar (**)` can't evaluate to zero directories. This isn't standard, as it should be `zero or more`.

**Comment by mrmlnc** [comment]
```js
'**/library/*/book.md'

micromatch
{
  re: /^(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)library\/(?!\.)(?=.)[^/]*?\/book\.md)$/
}

fast-glob.utils.pattern.makeRe
{
  re: /^(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)library\/(?!\.)(?=.)[^/]*?\/book\.md)$/
}

---

'{book.xml,**/library/*/book.md}'

micromatch
{
  re: /^(?:(book\.xml|(?:(?:(?!(?:^|\/)\.).)*?)\/library\/(?!\.)(?=.)[^/]*?\/book\.md))$/
}

fast-glob.utils.pattern.makeRe
{
  re: /^(?:(book\.xml|(?:(?:(?!(?:^|\/)\.).)*?)\/library\/(?!\.)(?=.)[^/]*?\/book\.md))$/
}
```

Well, looks like this is really a problem in the `micromatch` package.

```js
// {book.xml,**/library/*/book.md}
/^(?:(book\.xml|(?:(?:(?!(?:^|\/)\.).)*?)\/library\/(?!\.)(?=.)[^/]*?\/book\.md))$/
// **/library/*/book.md
/^(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)library\/(?!\.)(?=.)[^/]*?\/book\.md)$/
```

A little later I will create an issue in micromatch and try to find the reason for the current behavior.

**Comment by AgentEnder** [comment]
@mrmlnc Is there an upstream issue link for this? I'm trying to keep an eye on this. If there is any way that I can help out, let me know. This issue currently presents in Nx when we combine glob patterns for finding plugin project files.

**Comment by AgentEnder** [comment]
I repro'd on the picomatch repo and opened an issue there, hoping to accelerate this one too. Let me know if there is anything else from my side that could help with this.

**Comment by mrmlnc** [comment]
Already fixed in the master by #395 in this package. Will be shipped with [`3.3.0`](https://github.com/mrmlnc/fast-glob/issues/396).

**Comment by AgentEnder** [comment]
Awesome! Thanks for the fix @mrmlnc! Happy maintainers month, hoping things are well for you.
